src/internal/m365/exchange/events_restore.go

```diff
@@ -102,8 +108,115 @@ func (h eventRestoreHandler) restore(
 		return nil, clues.Stack(err)
 	}
 
+	// Fix up event instances in case we have a recurring event
+	err = h.updateRecurringEvents(ctx, userID, destinationID, ptr.Val(item.GetId()), event)
+	if err != nil {
+		return nil, clues.Stack(err)
+	}
+
 	info := api.EventInfo(event)
 	info.Size = int64(len(body))
 
 	return info, nil
 }
+
+func (h eventRestoreHandler) updateRecurringEvents(
+	ctx context.Context,
+	userID, containerID, eventID string,
```

Comment:
Nit, for standard naming conventions.
```suggestion
	userID, containerID, itemID string,
```